pkg/asset/agent/image/ignition.go

```diff
@@ -135,8 +136,20 @@ func (a *Ignition) Generate(dependencies asset.Parents) error {
 	if infraEnv.Spec.CpuArchitecture != "" {
 		archName = infraEnv.Spec.CpuArchitecture
 	}
-
-	releaseImageList, err := releaseImageList(agentManifests.ClusterImageSet.Spec.ReleaseImage, archName)
+	// Examine the release payload to see if its multi
+	releaseArch, err := agentcommon.DetermineReleaseImageArch(agentManifests.GetPullSecretData(), agentManifests.ClusterImageSet.Spec.ReleaseImage)
+	if err != nil {
+		logrus.Warnf("Unable to validate the release image architecture, using infraEnv.Spec.CpuArchitecture for the release image arch")
+		releaseArch = archName
```

Comment:
Do you know what happens inside assisted-service if we tell it we're giving it a single-arch image, but the actual image spec is for a multi-arch image? Does it just handle that transparently?

Also, this fallback won't work correctly for heterogeneous clusters (which require the `multi` release payload)... I assume that's because we're not planning to support those yet?

Reply:
I tested this, and was transparent on both x86 and arm. WRT heterogeneous installs, that will be a completely different effort. This PR is to ensure that the multi payload behaves the same way as a single arch payload.

Reply:
ack